README.md

```diff
@@ -2186,6 +2186,165 @@ $ m . /example
     [INFO] Path contents are:
     packages.nix  sources.json  sources.nix
 ```
+
+### Node Package Manager (NPM)
+
+#### makeNodeInterpreter
+
+Get a specific [Node.js][NODE_JS] version.
+Useful for passing a node version
+to either `makeNodeEnvironment`
+or `makeNodeModules`.
+
+Types:
+
+- makeNodeInterpreter (`function version -> package`):
+
+    - version (`enum [ "10" "12" "14" "16" ]`):
+      [Node.js][NODE_JS] version to use.
+
+Example:
+
+```nix
+# /path/to/my/project/makes/example/main.nix
+{ makeScript
+, makeNodeInterpreter
+, ...
+}:
+makeScript {
+  entrypoint = ''
+    node --version
+  '';
+  name = "example";
+  searchPaths = {
+    bin = [ (makeNodeInterpreter "16") ];
+  };
+}
+```
+
+```bash
+$ m . /example
+
+    v16.2.0
+```
+
+#### makeNodeModules
+
+Build a `node_modules` environment
+for a list of [NPM][NPM] dependencies.
+
+Types:
+
+- makeNodeModules (`function { ... } -> package`):
+
+    - name (`str`):
+      Custom name to assign to the build step, be creative, it helps in debugging.
+    - node (`package`):
+      [Node.js][NODE_JS] package
+      to run `npm install`.
+    - searchPaths (`asIn makeSearchPaths`): Optional.
+      Arguments here will be passed as-is to `makeSearchPaths`.
+      Defaults to `makeSearchPaths`'s defaults.
+    - dependencies (`listOf str`):
+      Direct dependencies to install.
+      Equivalent to `dependencies` in `package.json`.
+    - subDependencies (`listOf str`):
+      Inherited dependencies.
+      Relevant for completely pinning your environment.
+
+Example:
+
+```nix
+# /path/to/my/project/makes/example/main.nix
+{ makeScript
+, makeNodeInterpreter
+, makeNodeModules
+, ...
+}:
+let
+  hello = makeNodeModules {
+    name = "hello-world-npm";
+    node = makeNodeInterpreter "16";
+    dependencies = [ "hello-world-npm@1.1.1" ];
+    subDependencies = [];
+  };
+in
+makeScript {
+  replace = {
+    __argHello__ = hello;
+  };
+  entrypoint = ''
+    ls __argHello__
+  '';
+  name = "example";
+}
+```
+
+```bash
+$ m . /example
+
+    node_modules  package.json  package-lock.json  requirements
```

Comment:
this should be returning the node_modules/* instead of the parent dir of node_modules

Reply:
Will implement this in another MR.